packages/cli/src/admin-sdk.ts

```diff
@@ -0,0 +1,218 @@
+import { createClient } from './autogen/client/createClient';
+import fse from 'fs-extra';
+import { kebabCase, omit } from 'lodash';
+import chalk from 'chalk';
+import { readAsJson, getFiles, getDirectories, replaceField } from './utils';
+import cliProgress from 'cli-progress';
+import { createHash } from 'crypto';
+import traverse from 'traverse';
+
+const multibar = new cliProgress.MultiBar(
+  {
+    clearOnComplete: false,
+    hideCursor: true,
+    format: '|{bar}| {name} | {value}/{total}',
+  },
+  cliProgress.Presets.shades_grey
+);
+
+// todo change this to prod
+const root = 'https://qa.builder.io';
+
+export const importSpace = async (privateKey: string, directory: string, debug = false) => {
+  const graphqlClient = createClient({
+    fetcher: ({ query, variables }, fetch, qs) =>
+      fetch(`${root}/api/v2/admin?${qs.stringify({ query, variables })}`, {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${privateKey}`,
+        },
+      }).then(r => r.json()),
+  });
+
+  const spaceProgress = multibar.create(1, 0);
+  spaceProgress.start(1, 0, { name: 'getting space settings' });
+
+  try {
+    const space = await graphqlClient.chain.query
+      .downloadClone()
+      .execute({
+        models: { name: true, everything: true, content: true },
+        settings: true,
+        meta: true,
+      });
+    spaceProgress.setTotal(space.models.length);
+    let message = '';
+    fse.outputFileSync(
+      `./${directory}/settings.json`,
+      JSON.stringify({ ...space.settings, cloneInfo: space.meta }, undefined, 2)
+    );
+    space.models.forEach((model, index) => {
+      message += `${model.everything.name}${index !== space.models.length - 1 ? ',' : ''}`;
+      spaceProgress.update(index + 1, { name: message });
+      const { content, everything } = model;
+      // todo why conent is in everything
+      const { content: _, ...schema } = everything;
+      const modelName = kebabCase(model.name);
+      fse.emptyDirSync(`./${directory}/${modelName}`);
+      fse.outputFileSync(
```

Comment:
I like to use a package `fs-extra-promise` which lets you use async/await for much better perf and parallelization

```ts
await fse.emptyDirAsync(`./${directory}/${modelName}`);
```

etc

Reply:
can make a big difference when doing things like this that are in a loop/forEach so you don't block the thread

Reply:
just note at times you'll need to use Promise.all too, e.g. above

```ts
// from
spaces.modes.forEach(...)
// to
await Promise.all(spaces.models.map(async ...))
```